src/Functions/CRC.cpp

```diff
@@ -51,6 +51,12 @@ struct CRC32IEEEImpl : public CRCImpl<UInt32, CRC32_IEEE>
     static constexpr auto name = "CRC32IEEE";
 };
 
+constexpr UInt32 CRC32_C = 0x82f63b78;
```

Comment:
Thanks. Two problems:

1. This is not hardware accelerated. While on x86_64, AArch64, we use hardware-accelerated versions internally.

2. As we use this hash function internally in many places, this makes algorithmic complexity attacks trivial. (The PR is still acceptable, but it makes the need for resolving this problem more evident)

Reply:
I don't understand what I need to do about it.

Reply:
Re-use `Hash.h`, `intHashCRC32`

Reply:
Then, test for trivial cases of algorithmic complexity attacks, e.g., doing group by over CRC32C values.
It should reproduce a major slowdown.

It will be a motivation to add a flag to enable/disable this function, similarly to `enable_hyperscan`.

Reply:
But the function is for `String` and it's implementation is different from `intHashCRC32`.

Reply:
You're right. But still, we need a hardware-accelerated implementation if it's available.

PS. What is your motivation for adding CRC32-C?

Reply:
> PS. What is your motivation for adding CRC32-C?

Some users in my company need it.

Reply:
Ok, this is reasonable. Let's add hardware acceleration, make experiments with algorithmic complexity attacks, and continue.